docs/source/concepts/ledger-model/ledger-integrity.rst

```diff
@@ -55,26 +61,148 @@ below depicts the resulting order on the paint offer example:
 
-#. either `act` is itself **Create c** or a **Create c** happens before `act`
-#. `act` does not happen before any **Create c** action
-#. `act` does not happen after any exercise consuming `c`.
+.. _da-model-contract-consistency:
+
+Contract consistency
+````````````````````
+
+Contract consistency ensures that contracts are used after they have been created and before they are consumed.
+
+.. _def-contract-consistency:
+
+Definition »contract consistency«
+  A ledger is **consistent for a contract c** if all of the following holds for all actions `act` on `c`:
+
+  #. either `act` is itself **Create c** or a **Create c** happens before `act`
+  #. `act` does not happen before any **Create c** action
+  #. `act` does not happen after any exercise consuming `c`.
 
-A ledger is **consistent** if it is consistent for all contracts.
 
 The consistency condition rules out the double spend example.
 As the red path below indicates, the second exercise in the example happens after a consuming exercise on the same
-contract, violating the consistency criteria.
+contract, violating the contract consistency criteria.
 
 .. https://www.lucidchart.com/documents/edit/c6113536-70f4-42a4-920d-3c9497f8f7c4
 .. image:: ./images/consistency-banning-double-spends.svg
    :align: center
    :width: 100%
 
+
+.. _def-contract-state:
+
+In addition to the consistency notions, the before-after relation on actions can also be used to define the notion of
+**contract state** at any point in a given transaction.
+The contract state is changed by creating the contract and by exercising it consumingly.
+At any point in a transaction, we can then define the latest state change in the obvious way.
+Then, given a point in a transaction, the contract state of `c` is:
+
+#. **active**, if the latest state change of `c` was a create;
+
+#. **archived**, if the latest state change of `c` was a consuming exercise;
+
+#. **inexistent**, if `c` never changed state.
+
+A ledger is consistent for `c` exactly if **Exercise** and **Fetch** actions on `c` happen only when `c` is active,
+and **Create** actions only when `c` is inexistent.
+The figures below visualize the state of different contracts at all points in the example ledger.
+
+.. https://www.lucidchart.com/documents/edit/19226d95-e8ba-423a-8546-e5bae6bd3ab7
+.. figure:: ./images/consistency-paint-offer-activeness.svg
+   :align: center
+   :width: 100%
+
+   Activeness of the `PaintOffer` contract
+
+.. https://www.lucidchart.com/documents/edit/19226d95-e8ba-423a-8546-e5bae6bd3ab7
+.. figure:: ./images/consistency-alice-iou-activeness.svg
+   :align: center
+   :width: 100%
+
+   Activeness of the `Iou Bank A` contract
+
+The notion of order can be defined on all the different ledger structures: actions, transactions, lists of transactions,
+and ledgers.
+Thus, the notions of (internal) activeness consistency, inputs and outputs, and contract state can also all be defined on all these
```

Comment:
Internal is undefined at this point; also, "activeness consistency" is not defined anywhere (it appears also lower in the text)

Reply:
Good catch. Removed "internal" and "activeness".